rest/api_test.go

```diff
@@ -7918,3 +7918,722 @@ func TestDocChannelSetPruning(t *testing.T) {
+
+		// Retrieve the attachment added to the document.
+		actualAttBody := retrieveAttachment(docID, att1Name)
+		require.Equal(t, att1Body, actualAttBody)
+
+		// Get the document and check the attachment metadata.
+		attachments := retrieveAttachmentMeta(docID)
+		require.Len(t, attachments, 1)
+		meta, ok := attachments[att1Name].(map[string]interface{})
+		require.True(t, ok)
+		assert.True(t, meta["stub"].(bool))
+		assert.Equal(t, "content/type", meta["content_type"].(string))
+		assert.Equal(t, "sha1-5vJRip1gGo8YsI9yEJmmv6DabXk=", meta["digest"].(string))
+		assert.Equal(t, float64(40), meta["length"].(float64))
+		assert.Equal(t, float64(2), meta["revpos"].(float64))
+		assert.Equal(t, float64(2), meta["ver"].(float64))
+
+		// Retrieve the key used for internal attachment storage and retrieval.
+		att1Key := retrieveAttachmentKey(docID, att1Name)
+		require.NotEmpty(t, att1Key)
+
+		// Add another attachment to the same document.
+		att2Name := "bob.txt"
+		att2Body := "this is the body of attachment bob.txt"
+		revID = storeAttachment(docID, revID, att2Name, att2Body)
+		require.Equal(t, "3-9d8d0bf2e87982d97356a181a693291b", revID)
+
+		// Retrieve the second attachment added to the document.
+		actualAtt2Body := retrieveAttachment(docID, att2Name)
+		require.Equal(t, att2Body, actualAtt2Body)
+
+		// Get the document and check the attachment metadata.
+		attachments = retrieveAttachmentMeta(docID)
+		require.Len(t, attachments, 2)
+
+		meta, ok = attachments[att1Name].(map[string]interface{})
+		require.True(t, ok)
+		assert.True(t, meta["stub"].(bool))
+		assert.Equal(t, "content/type", meta["content_type"].(string))
+		assert.Equal(t, "sha1-5vJRip1gGo8YsI9yEJmmv6DabXk=", meta["digest"].(string))
+		assert.Equal(t, float64(40), meta["length"].(float64))
+		assert.Equal(t, float64(2), meta["revpos"].(float64))
+		assert.Equal(t, float64(2), meta["ver"].(float64))
+
+		meta, ok = attachments[att2Name].(map[string]interface{})
+		require.True(t, ok)
+		assert.True(t, meta["stub"].(bool))
+		assert.Equal(t, "content/type", meta["content_type"].(string))
+		assert.Equal(t, "sha1-3oMVZvHjOQkkEK7K/xp0tqkuj1Q=", meta["digest"].(string))
+		assert.Equal(t, float64(38), meta["length"].(float64))
+		assert.Equal(t, float64(3), meta["revpos"].(float64))
+		assert.Equal(t, float64(2), meta["ver"].(float64))
+
+		// Retrieve the key used for internal attachment storage and retrieval.
+		att2Key := retrieveAttachmentKey(docID, att2Name)
+		require.NotEmpty(t, att2Key)
+		require.NotEqual(t, att1Key, att2Key)
+
+		// Purge the entire document.
+		rt.purgeDoc(docID)
+
+		// Check whether both attachments are removed from the underlying storage.
+		requireAttachmentNotFound(docID, att1Name)
+		rt.requireDocNotFound(att1Key)
+		requireAttachmentNotFound(docID, att2Name)
+		rt.requireDocNotFound(att2Key)
+	})
+
+	t.Run("single inline attachment removal upon document update", func(t *testing.T) {
```

Comment:
Is there a difference between the inline and non-inline case, once the document has been created, other than the first case has one revision and the second has two?

Reply:
There is no difference between the inline and non-inline case, once the document has been created, other than the first case has one revision and the second has two. However, this unit test has been included as an end-to-end coverage for inline flow.